ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -684,6 +689,21 @@ class JITModBuilder {
       FailBuilder.CreateStore(
           ConstantInt::get(PointerSizedIntTy, LiveVals.size()), GEP);
 
+      // Store the stackmap address and length in a separate struct to save
+      // arguments.
+      AllocaInst *StackMapStruct = FailBuilder.CreateAlloca(
+          VecLenStructTy, ConstantInt::get(PointerSizedIntTy, 1));
+      GEP = FailBuilder.CreateGEP(VecLenStructTy, StackMapStruct,
+                                  {ConstantInt::get(PointerSizedIntTy, 0),
+                                   ConstantInt::get(Int32Ty, 0)});
```

Comment:
Why `Int32Ty`?

Reply:
I was curious.

According to the [docs](https://llvm.org/docs/LangRef.html#i-getelementptr):
> When indexing into an array, pointer or vector, integers of any width are allowed, and they are not required to be constant.

So may as well use the smallest type that can express the constant (here `i1`)?

Reply:
No, no, no :P

This is a struct, so i read the wrong bit:

> When indexing into a (optionally packed) structure, only i32 integer constants are allowed (when using a vector of indices they must all be the same i32 integer constant).

Reply:
Aha!